tests/virtualization/universal/prepare_guests.pm

```diff
@@ -15,30 +15,56 @@ use warnings;
+use File::Copy 'copy';
+use File::Path 'make_path';
+
+sub create_profile {
+    my ($vm_name, $arch, $mac, $ip) = @_;
+    my $version = $vm_name =~ /sp/ ? $vm_name =~ s/\D*(\d+)sp(\d)\D*/$1.$2/r : $vm_name =~ s/\D*(\d+)\D*/$1/r;
+    my $path = $version >= 15 ? "virtualization/autoyast/guest_15.xml.ep" : "virtualization/autoyast/guest_12.xml.ep";
+    my $scc_code = get_required_var("SCC_REGCODE");
+    my %ltss_products = @{get_var_array("LTSS_REGCODES_SECRET")};
+    my $ca_str = "SLE_" . $version =~ s/\./_SP/r;
+    record_info("$ca_str");
+    my $profile = get_test_data($path);
+    $profile =~ s/\{\{GUEST\}\}/$vm_name/g;
+    $profile =~ s/\{\{SCC_REGCODE\}\}/$scc_code/g;
+    $profile =~ s/\{\{ARCH\}\}/$arch/g;
+    $profile =~ s/\{\{MAC\}\}/$mac/g;
+    $profile =~ s/\{\{IP\}\}/$ip/g;
+    $profile =~ s/\{\{VERSION\}\}/$version/g;
+    $profile =~ s/\{\{CA_STR\}\}/$ca_str/g;
+    my $host_os_version = get_var('DISTRI') . "s" . lc(get_var('VERSION') =~ s/-//r);
+    my $incident_repos = "";
+    $incident_repos = get_var('INCIDENT_REPO', '') if ($vm_name eq $host_os_version || $vm_name eq "${host_os_version}PV" || $vm_name eq "${host_os_version}HVM");
+    my $vars = {
+        vm_name => $vm_name,
+        ltss_code => $ltss_products{$version},
+        repos => [split(/,/, $incident_repos)]
+    };
+    my $output = Mojo::Template->new(vars => 1)->render($profile, $vars);
+    save_tmp_file("$vm_name.xml", $output);
+    # Copy profile to ulogs directory, so profile is available in job logs
+    make_path('ulogs');
+    copy(hashed_string("$vm_name.xml"), "ulogs/$vm_name.xml");
+    return autoinst_url . "/files/$vm_name.xml";
+}
 
 sub run {
     my $self = shift;
     # Use serial terminal, unless defined otherwise. The unless will go away once we are certain this is stable
-    $self->select_serial_terminal unless get_var('_VIRT_SERIAL_TERMINAL', 1) == 0;
-    # Clean up VMs if any. This makes it possable to reinstall VMs.
+    #    $self->select_serial_terminal unless get_var('_VIRT_SERIAL_TERMINAL', 1) == 0;
+    select_console('root-console');
     systemctl("restart libvirtd");
     assert_script_run('for i in $(virsh list --name|grep sles);do virsh destroy $i;done');
     assert_script_run('for i in $(virsh list --name --inactive); do virsh undefine $i --remove-all-storage;done');
 
+
     # Ensure additional package is installed
-    zypper_call '-t in libvirt-client iputils nmap supportutils xmlstarlet';
+    zypper_call '-t in libvirt-client iputils nmap supportutils';
 
     assert_script_run "mkdir -p /var/lib/libvirt/images/xen/";
 
-    if (is_sle('<=12-SP1')) {
-        script_run "umount /home";
-        assert_script_run qq(sed -i 's/\\/home/\\/var\\/lib\\/libvirt\\/images\\/xen/g' /etc/fstab);
-        script_run "mount /var/lib/libvirt/images/xen/";
-    }
-
-    assert_script_run "curl -f -v " . data_url("virt_autotest/libvirtd.conf") . " >> /etc/libvirt/libvirtd.conf";
-    systemctl 'restart libvirtd';
```

Comment:
I can understand L33-37 can be removed because 12sp1 is out of support now. But for L39-40, I guess there are some needed changes in libvirtd.conf which is different from default config. Why just remove this part?

Reply:
oh, okay.